### PR TITLE
test: add Capacity Buffers integ tests

### PR DESCRIPTION
**Pull request by sumukha-radhakrishna**
**Description**

### Summary

  Adds KWOK integration tests for the CapacityBuffer feature, validating end-to-end behavior
  across provisioning, disruption, lifecycle, and edge cases.

  Tests

  PodTemplateRef Provisioning (2 tests)
  - Provisions capacity and reaches Provisioning=True/FitsExistingCapacity
  - Updates buffer status when PodTemplate is updated (generation tracking)

  ScalableRef Provisioning (5 tests)
  - Provisions with percentage (20% of 10 = 2 replicas)
  - Provisions with fixed replicas
  - Recovers when scalable ref is created after buffer (late binding)
  - Provisions with fixed replicas via scalableRef
  - Recalculates percentage when deployment scales (5→20 replicas)

  Consumer Interaction (2 tests)
  - Consumer pods use existing buffer capacity without new NodeClaims (30s consistency check)
  - Buffer refills after consumption (3 CPU pods force genuine node-level contention)

  Disruption (3 tests)
  - Buffer nodes not empty-consolidated (60s consistency window)
  - Buffer nodes consolidated after buffer deletion
  - Drift replaces buffer nodes and buffer re-satisfies

  NodePool Limits (1 test)
  - Buffer cannot provision beyond NodePool node count limit

  Multiple Buffers (1 test)
  - Two buffers with different templates provision independently

  Edge Cases (3 tests)
  - Zero replicas when limits < single pod request
  - No leaked nodes on rapid create/delete (verifies provisioning actually occurs first)
  - Growing buffer replicas when limits are increased

**How was this change tested?**
NA

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/pkg/environment/common/setup.go
Comment (DerekFrank) [inline test/pkg/environment/common/setup.go]: Completely unrelated to this PR, but it really bugs me that this isn't alphabetical and grouped by *. prefix

File: test/suites/regression/capacitybuffer_test.go [hunk 1]
```diff
@@ -0,0 +1,1052 @@
...
+				cond := findBufferCondition(cb.Status.Conditions, autoscalingv1alpha1.ProvisioningCondition)
+				g.Expect(cond).ToNot(BeNil())
+				g.Expect(cond.Status).To(Equal(metav1.ConditionTrue))
+			}).WithTimeout(2 * time.Minute).Should(Succeed())
+		})
+	})
+
+	Context("Disruption", func() {
+		It("should not empty-consolidate nodes hosting buffer pods", func() {
+			buffer := &autoscalingv1alpha1.CapacityBuffer{
```
Comment (DerekFrank) [inline test/suites/regression/capacitybuffer_test.go]: Can we pull these out into a test.CapacityBuffer(...overrides) like we do for pods, nodeclaims and nodes?

File: test/suites/regression/capacitybuffer_test.go [hunk 2]
```diff
@@ -0,0 +1,1052 @@
...
+				Spec: autoscalingv1alpha1.CapacityBufferSpec{
+					ProvisioningStrategy: lo.ToPtr(autoscalingv1alpha1.ActiveProvisioningStrategy),
+					PodTemplateRef:       &autoscalingv1alpha1.LocalObjectRef{Name: "buffer-template"},
+					Replicas:             lo.ToPtr(int32(3)),
+				},
+			}
+
+			env.ExpectCreated(bufferTemplate, buffer)
+
+			Eventually(func(g Gomega) {
```
Comment (DerekFrank) [inline test/suites/regression/capacitybuffer_test.go]: Can we pull these out into a EventuallyExpectCapacityBufferProvisioned and EventuallyExpectCapacityBufferReplicas? That way we can cut the tests that just check that, all of the tests that use CapacityBuffers can just check that as they use them

Comment (DerekFrank) [reply test/suites/regression/capacitybuffer_test.go]: Do you think we also need the reverse? Whatever status we put on the buffer when we can't provision it

File: test/pkg/environment/common/setup.go
Comment (sumukha-radhakrishna) [reply test/pkg/environment/common/setup.go]: Can group these

File: test/suites/regression/capacitybuffer_test.go (hunk 2 above)
Comment (sumukha-radhakrishna) [reply test/suites/regression/capacitybuffer_test.go]: Yeah lm move these to common functions

**Review by DerekFrank — APPROVED** [review]
/lgtm
